diff -Naur Xfoil/bin/CMakeLists.txt Xfoil.patched/bin/CMakeLists.txt
--- Xfoil/bin/CMakeLists.txt	1969-12-31 21:00:00.000000000 -0300
+++ Xfoil.patched/bin/CMakeLists.txt	2021-02-07 19:07:43.920065649 -0300
@@ -0,0 +1,153 @@
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -fdefault-real-8 -fallow-argument-mismatch")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2 -DUNDERSCORE")
+set(ROOT ${CMAKE_CURRENT_SOURCE_DIR}/..)
+set(SOURCE ${ROOT}/src)
+set(OSRC ${ROOT}/osrc)
+set(PLTLIB ${X11_LIBRARIES})
+set(FTNLIB ${LDFLAGS})
+
+message(STATUS "PLTLIB=${PLTLIB}")
+message(STATUS "FTNLIB=${FTNLIB}")
+enable_language(C)
+add_library(getosfile OBJECT ${OSRC}/getosfile.c)
+
+enable_language(Fortran)
+# pxplot objects
+add_library(plutil OBJECT ${SOURCE}/plutil.f)
+add_library(gui OBJECT ${SOURCE}/gui.f)
+add_library(userio OBJECT ${SOURCE}/userio.f)
+
+# pplot objects
+add_library(polplt OBJECT ${SOURCE}/polplt.f ${SOURCE}/PINDEX.INC)
+add_library(sort OBJECT ${SOURCE}/sort.f)
+add_library(iopol OBJECT ${SOURCE}/iopol.f ${SOURCE}/PINDEX.INC)
+
+# xfoil objects
+add_library(xpanel OBJECT ${SOURCE}/xpanel.f ${SOURCE}/XFOIL.INC)
+add_library(xoper OBJECT ${SOURCE}/xoper.f ${SOURCE}/XFOIL.INC ${SOURCE}/BLPAR.INC)
+add_library(xtcam OBJECT ${SOURCE}/xtcam.f ${SOURCE}/XFOIL.INC ${SOURCE}/XDES.INC)
+add_library(xgdes OBJECT ${SOURCE}/xgdes.f ${SOURCE}/XFOIL.INC ${SOURCE}/XDES.INC)
+add_library(xqdes OBJECT ${SOURCE}/xqdes.f ${SOURCE}/XFOIL.INC ${SOURCE}/XDES.INC)
+add_library(xmdes OBJECT ${SOURCE}/xmdes.f ${SOURCE}/XFOIL.INC ${SOURCE}/XDES.INC ${SOURCE}/CIRCLE.INC)
+add_library(xsolve OBJECT ${SOURCE}/xsolve.f ${SOURCE}/XFOIL.INC)
+add_library(xbl OBJECT ${SOURCE}/xbl.f ${SOURCE}/XFOIL.INC ${SOURCE}/XBL.INC ${SOURCE}/BLPAR.INC)
+add_library(xblsys OBJECT ${SOURCE}/xblsys.f ${SOURCE}/XBL.INC ${SOURCE}/BLPAR.INC)
+add_library(xpol OBJECT ${SOURCE}/xpol.f ${SOURCE}/XFOIL.INC ${SOURCE}/PINDEX.INC)
+add_library(xplots OBJECT ${SOURCE}/xplots.f ${SOURCE}/XFOIL.INC ${SOURCE}/XDES.INC)
+add_library(pntops OBJECT ${SOURCE}/pntops.f ${SOURCE}/XFOIL.INC ${SOURCE}/XDES.INC)
+add_library(xgeom OBJECT ${SOURCE}/xgeom.f)
+add_library(xutils OBJECT ${SOURCE}/xutils.f)
+add_library(modify OBJECT ${SOURCE}/modify.f)
+add_library(blplot OBJECT ${SOURCE}/blplot.f ${SOURCE}/XFOIL.INC ${SOURCE}/BLPAR.INC)
+add_library(aread OBJECT ${SOURCE}/aread.f)
+add_library(naca OBJECT ${SOURCE}/naca.f)
+add_library(spline_obj OBJECT ${SOURCE}/spline.f)
+add_library(dplot OBJECT ${SOURCE}/dplot.f ${SOURCE}/XFOIL.INC)
+add_library(profil OBJECT ${SOURCE}/profil.f)
+add_library(frplot OBJECT ${SOURCE}/frplot.f)
+add_library(ntcalc OBJECT ${SOURCE}/ntcalc.f)
+add_library(osmap OBJECT ${OSRC}/osmap.f)
+
+set(
+        PXPLOTOBJ
+        $<TARGET_OBJECTS:plutil>
+        $<TARGET_OBJECTS:gui>
+)
+set(
+        XUTILOBJ
+        $<TARGET_OBJECTS:userio>
+)
+set(
+        PPLOTOBJ
+        $<TARGET_OBJECTS:polplt>
+        $<TARGET_OBJECTS:sort>
+        $<TARGET_OBJECTS:iopol>
+)
+set(
+        XFOILOBJ
+        $<TARGET_OBJECTS:xpanel>
+        $<TARGET_OBJECTS:xoper>
+        $<TARGET_OBJECTS:xtcam>
+        $<TARGET_OBJECTS:xgdes>
+        $<TARGET_OBJECTS:xqdes>
+        $<TARGET_OBJECTS:xmdes>
+        $<TARGET_OBJECTS:xsolve>
+        $<TARGET_OBJECTS:xbl>
+        $<TARGET_OBJECTS:xblsys>
+        $<TARGET_OBJECTS:xpol>
+        $<TARGET_OBJECTS:xplots>
+        $<TARGET_OBJECTS:pntops>
+        $<TARGET_OBJECTS:xgeom>
+        $<TARGET_OBJECTS:xutils>
+        $<TARGET_OBJECTS:modify>
+        $<TARGET_OBJECTS:blplot>
+        $<TARGET_OBJECTS:polplt>
+        $<TARGET_OBJECTS:aread>
+        $<TARGET_OBJECTS:naca>
+        $<TARGET_OBJECTS:spline_obj>
+        $<TARGET_OBJECTS:plutil>
+        $<TARGET_OBJECTS:iopol>
+        $<TARGET_OBJECTS:gui>
+        $<TARGET_OBJECTS:sort>
+        $<TARGET_OBJECTS:dplot>
+        $<TARGET_OBJECTS:profil>
+)
+set(
+        OSOBJ
+        $<TARGET_OBJECTS:frplot>
+        $<TARGET_OBJECTS:ntcalc>
+        $<TARGET_OBJECTS:osmap>
+        $<TARGET_OBJECTS:getosfile>
+)
+set(
+        PLTOBJ
+        libPlt
+)
+
+# Creating final executables
+add_executable(
+        pxplot ${SOURCE}/pxplot.f ${SOURCE}/PXPLOT.INC
+        ${PXPLOTOBJ}
+        ${XUTILOBJ}
+)
+target_link_libraries(
+        pxplot
+        ${PLTLIB}
+        ${FTNLIB}
+        ${PLTOBJ}
+)
+set_target_properties(pxplot PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${PKGDIR})
+
+add_executable(
+        pplot ${SOURCE}/pplot.f
+        ${PPLOTOBJ}
+        ${XUTILOBJ}
+)
+target_link_libraries(
+        pplot
+        ${PLTLIB}
+        ${FTNLIB}
+        ${PLTOBJ}
+)
+set_target_properties(pplot PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${PKGDIR})
+
+add_executable(
+        xfoil ${SOURCE}/xfoil.f
+        ${XFOILOBJ}
+        ${XUTILOBJ}
+        ${OSOBJ}
+)
+target_link_libraries(
+        xfoil
+        ${PLTLIB}
+        ${FTNLIB}
+        ${PLTOBJ}
+)
+set_target_properties(xfoil PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${PKGDIR})
+
+add_dependencies(xfoil osgen)
+
+file(
+        INSTALL ${ROOT}/xfoil_doc.txt
+        DESTINATION ${CMAKE_BINARY_DIR}/share/xfoil/doc/
+)
diff -Naur Xfoil/CMakeLists.txt Xfoil.patched/CMakeLists.txt
--- Xfoil/CMakeLists.txt	1969-12-31 21:00:00.000000000 -0300
+++ Xfoil.patched/CMakeLists.txt	2021-02-06 22:41:43.036069505 -0300
@@ -0,0 +1,7 @@
+FIND_PACKAGE(X11 REQUIRED)
+add_subdirectory(orrs/bin)
+add_subdirectory(plotlib)
+add_subdirectory(bin)
+
+
+
diff -Naur Xfoil/orrs/bin/CMakeLists.txt Xfoil.patched/orrs/bin/CMakeLists.txt
--- Xfoil/orrs/bin/CMakeLists.txt	1969-12-31 21:00:00.000000000 -0300
+++ Xfoil.patched/orrs/bin/CMakeLists.txt	2021-02-07 13:27:28.144055116 -0300
@@ -0,0 +1,37 @@
+set(ORRS ${CMAKE_CURRENT_SOURCE_DIR}/..)
+set(ORRS_ROOT ${CMAKE_CURRENT_SOURCE_DIR})
+set(SOURCE ${ORRS}/src)
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -fdefault-real-8")
+
+message(STATUS "ORRS=${ORRS}")
+message(STATUS "ORRS_ROOT=${ORRS_ROOT}")
+enable_language(Fortran)
+
+# Creating libraries needed by osgen
+add_library(io OBJECT ${SOURCE}/io.f)
+add_library(orrs_spline OBJECT ${SOURCE}/spline.f)
+add_library(getarg0 OBJECT ${SOURCE}/getarg0.f)
+add_library(orrs_osmap OBJECT ${SOURCE}/osmap.f)
+set(
+        ORRS_OBJS
+        $<TARGET_OBJECTS:io>
+        $<TARGET_OBJECTS:orrs_spline>
+        $<TARGET_OBJECTS:getarg0>
+        $<TARGET_OBJECTS:orrs_osmap>
+)
+
+add_executable(
+        osgen ${SOURCE}/osgen.f
+        ${ORRS_OBJS}
+)
+target_link_libraries(osgen ${LDFLAGS})
+
+# Certifies the directory exists otherwise rename will give a No such file or directory error
+file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/share/xfoil/orrs)
+add_custom_command(
+        TARGET osgen POST_BUILD
+        COMMAND osgen osmaps_ns.lst
+        COMMAND ${CMAKE_COMMAND} -E rename osmap.dat ${CMAKE_BINARY_DIR}/share/xfoil/orrs/osmap.dat
+        WORKING_DIRECTORY ${ORRS}
+        COMMENT "Creating osmap.dat and moving it ${ORRS}/osmap.dat to ${CMAKE_BINARY_DIR}/share/xfoil/orrs/"
+)
diff -Naur Xfoil/plotlib/CMakeLists.txt Xfoil.patched/plotlib/CMakeLists.txt
--- Xfoil/plotlib/CMakeLists.txt	1969-12-31 21:00:00.000000000 -0300
+++ Xfoil.patched/plotlib/CMakeLists.txt	2021-02-07 13:40:33.777402902 -0300
@@ -0,0 +1,44 @@
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -fno-range-check -fdefault-real-8 -std=legacy")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2 -DUNDERSCORE")
+set(INSTALLDIR ${CMAKE_SOURCE_DIR}/plotlib/)
+
+message(STATUS "INSTALLDIR=${INSTALLDIR}")
+message(STATUS "X11_LIBRARIES=${X11_LIBRARIES}")
+
+# Xwin2 requires X11 to link properly
+enable_language(C)
+add_library(Xwin2 OBJECT Xwin2.c)
+link_libraries(Xwin2 ${X11_LIBRARIES})
+
+enable_language(Fortran)
+add_library(plt_base OBJECT plt_base.f pltlib.inc)
+add_library(plt_font OBJECT plt_font.f CHAR.INC SLAN.INC MATH.INC SYMB.INC)
+add_library(plt_util OBJECT plt_util.f)
+add_library(plt_color OBJECT plt_color.f pltlib.inc)
+add_library(set_subs OBJECT set_subs.f pltlib.inc)
+add_library(gw_subs OBJECT gw_subs.f pltlib.inc)
+add_library(ps_subs OBJECT ps_subs.f pltlib.inc)
+add_library(plt_old OBJECT plt_old.f pltlib.inc)
+add_library(plt_3D OBJECT plt_3D.f)
+
+# Setting variables to create libPlt.a
+set(
+    OBJ
+    $<TARGET_OBJECTS:plt_font>
+    $<TARGET_OBJECTS:plt_util>
+    $<TARGET_OBJECTS:plt_color>
+    $<TARGET_OBJECTS:set_subs>
+    $<TARGET_OBJECTS:gw_subs>
+    $<TARGET_OBJECTS:ps_subs>
+    $<TARGET_OBJECTS:Xwin2>
+)
+set(OBJOLD $<TARGET_OBJECTS:plt_old>)
+set(OBJ3D $<TARGET_OBJECTS:plt_3D>)
+
+# Create liblibPlt and change its name to libPlt
+add_library(libPlt STATIC plt_base.f pltlib.inc ${OBJ} ${OBJOLD} ${OBJ3D} ${OBJMISC})
+set_target_properties(libPlt PROPERTIES OUTPUT_NAME "Plt")
+add_custom_command(
+        TARGET libPlt
+        POST_BUILD COMMAND ${CMAKE_RANLIB} $<TARGET_FILE:libPlt>
+)
